test/test.ml

```diff
@@ -23,10 +23,12 @@ let log = Printf.ksprintf (fun s -> if !verbose then Printf.fprintf stdout "%s\n
 
 let tests =
   [ "foo bar baz", ["foo"; "bar"; "baz"];
-    "foo \"bar\" baz", ["foo"; "bar"; "baz"];
-    "f\\\ oo b\\\"r baz", ["f oo"; "b\"r"; "baz"];
-    "foo bar\"bie\"boo baz", ["foo"; "barbieboo"; "baz"];
-    "  ", []
+    {|foo "bar" baz|}, ["foo"; "bar"; "baz"];
+    {|f\ oo b\"r baz|}, ["f oo"; "b\"r"; "baz"];
+    {|foo bar"bie"boo baz|}, ["foo"; "barbieboo"; "baz"];
+    {|foo "bar baz"|}, ["foo"; "bar baz"];
+    {|foo bar\ baz|}, ["foo"; "bar baz"];
+    "  ", [];
```

Comment:
The below fails. We could keep track of the previous state and then unconditionally add if the previous state was `Quoted`. I'm not sure what the semantics were before this patch.
```suggestion
    "  ", [];
    {|some-empty-string: ""|}, ["some-empty-string:"; ""];
```

Reply:
With the latest release, the above test fails:
```
Error - failed to parse. Got:
'some-empty-string:'
Expected
'some-empty-string:',''
```

This means the code proposed in this PR and in main have the same behaviour in respect to your test case. Now, I'm not sure what the desired behaviour is -- but happy to discuss this in a separate issue / PR.

Reply:
Yes, I noted that above.